striker-ui/Makefile.am

```diff
@@ -1,136 +1,105 @@
-			components/Network/index.tsx \
-			components/Network/Network.tsx \
-			components/Network/processNetwork.ts \
-			components/CPU.tsx \
-			components/Bars/index.tsx \
-			components/Bars/ProgressBar.tsx \
-			components/Bars/AllocationBar.tsx \
-			pages/index.tsx \
-			pages/_app.tsx \
-			pages/_document.tsx \
+			types \
+			components \
+			pages \
 			.eslintrc.json \
 			commitlint.config.js \
 			.eslintignore \
+			.env \
 			package-lock.json
 
-htmldir			= $(localstatedir)/www/html
-outdir			= out
-
-# Trigger build target on make call without parameters.
-all: out
+# Blank target; the build output directory should be usable as-is.
+$(nextoutdir): ;
 
-# Note: the input file to the generate endpoint prefix step must exist in
-#       EXTRA_DIST.
+# This target is for maintainers only; do not execute during CI/CD.
 #
-out:
-	-@echo "Copying required build files to build (current) directory."
-	rsync -av --exclude "[Mm]ake*" $(srcdir)/ ./
-	-@echo "Allow tools to write to files in the build directory."
-	chmod -R +w .
+$(nodemodulesdir):
 	-@echo "Install node modules (dependencies) prior to building."
 	npm install --no-package-lock --ignore-scripts
-	-@echo "Generate endpoint prefix."
-	sed 's@=.*@=/cgi-bin@' <env.development >.env.local
-	-@echo "Build front-end project."
+
+# This target is for maintainers only; do not execute during CI/CD. In
+# addition, this target can only be used if the local git repository exists.
+#
+# Note: this target must **not** be renamed to the same name as the build
+# output directory because it will trigger a build during the `make` implicit
+# call.
+#
+.PHONY: build
+build: $(nodemodulesdir)
+	-@echo "Remove build output files to prepare for rebuild."
+	rm -rf $(nextoutdir) $(nextbuilddir)
+	-@echo "Build web UI."
 	npm run build
+	-@echo "Stage only the build output directory and commit the build output."
+	git reset \
+		&& git add $(nextoutdir) \
+		&& git commit -m "chore: rebuild web UI"
 
 install-data-hook:
 	-@echo "Place build output files."
-	cp -r $(outdir)/index.html $(outdir)/_next $(DESTDIR)/$(htmldir)/
+	cp -r --no-preserve=mode $(srcdir)/$(nextoutdir)/$(outindexfile) $(srcdir)/$(nextoutdir)/$(outjsmodulesdir) $(DESTDIR)/$(htmldir)/
 	-@echo "Create symlink to images to enable borrowing icon etc. without duplicating."
-	(cd $(DESTDIR)/$(htmldir); $(LN_S) skins/alteeve/images pngs)
+	(cd $(DESTDIR)/$(htmldir); $(LN_S) skins/alteeve/images $(outimagesdir))
 
 uninstall-hook:
 	-@echo "Remove all installed files of the current module."
-	(cd $(DESTDIR)/$(htmldir); rm -rf index.html _next pngs)
+	(cd $(DESTDIR)/$(htmldir); rm -rf $(outindexfile) $(outjsmodulesdir) $(outimagesdir))
 
 clean-local:
-	-@echo "Clean up build output files."
-	test $(srcdir) == . && rm -rf $(outdir) .next || find . -mindepth 1 -maxdepth 1 -exec rm -rf {} +
+	-@echo "Clean up node modules."
+	rm -rf $(nodemodulesdir)
 
 distclean-local: clean-local
+
+maintainer-clean-local: clean-local
```

Comment:
you don´t need either of maintainer-clean-local or distclean-local target.

make maintainer-clean or distclean automatically calls into make clean (and clean-local)

Reply:
Removed `distclean-local` and `maintainer-clean-local` targets as suggested.